### PR TITLE
CMAKE_OSX_ARCHITECTURES for MacOS cross compilation

### PR DESCRIPTION
**Pull request by Xarbirus**
A way to utilise the [CMAKE_OSX_ARCHITECTURES](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html) functionality to cross compile for MacOS.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -850,7 +850,8 @@ endif()
 
 set(ARCH_FLAGS "")
 
-if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64") OR ("${CMAKE_GENERATOR_PLATFORM_LWR}" MATCHES "arm64"))
+if ((APPLE AND CMAKE_OSX_ARCHITECTURES AND ("${CMAKE_OSX_ARCHITECTURES}" MATCHES "arm64")) OR
+    (((APPLE AND NOT CMAKE_OSX_ARCHITECTURES) OR NOT APPLE) AND (("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm") OR ("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "aarch64") OR ("${CMAKE_GENERATOR_PLATFORM_LWR}" MATCHES "arm64"))))
```
Comment (cebtenzzre) [inline CMakeLists.txt:854]: Any reason something more like this wouldn't work?
```suggestion
if (CMAKE_OSX_ARCHITECTURES MATCHES "arm64" OR CMAKE_GENERATOR_PLATFORM_LWR MATCHES "arm64" OR
    (NOT CMAKE_OSX_ARCHITECTURES AND NOT CMAKE_GENERATOR_PLATFORM_LWR AND
     (CMAKE_SYSTEM_PROCESSOR MATCHES "arm" OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")))
```

Or to at least use an if/else chain to compute whether we are building for ARM?

Comment (Xarbirus) [reply CMakeLists.txt]: Yes, this option works and looks much better.
Accepted the suggestion and updated the conditions for the second branch (x86).

File: CMakeLists.txt [hunk 2]
```diff
@@ -850,7 +850,9 @@ endif()
 
 set(ARCH_FLAGS "")
 
-if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64") OR ("${CMAKE_GENERATOR_PLATFORM_LWR}" MATCHES "arm64"))
+if (CMAKE_OSX_ARCHITECTURES MATCHES "arm64" OR CMAKE_GENERATOR_PLATFORM_LWR MATCHES "arm64" OR
+    (NOT CMAKE_OSX_ARCHITECTURES AND NOT CMAKE_GENERATOR_PLATFORM_LWR AND
+     (CMAKE_SYSTEM_PROCESSOR MATCHES "arm" OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")))
```
[the remark this answers is not in the document]
Comment (Xarbirus) [reply CMakeLists.txt]: I agree with everything except `CMAKE_SYSTEM_PROCESSOR`, this variable can be ​​arm or arm64. Updated both branches.

**Review by cebtenzzre — APPROVED** [review]

**Comment by cebtenzzre** [comment]
Oops, I've just realized that CMAKE_OSX_ARCHITECTURES is actually a list, this PR as merged is treating it like a single architecture string. How are we supposed to handle that?

**Comment by Xarbirus** [comment]
It seems to me that the best way to handle this situation is to separate determination of the architecture(s) and compile definitions adding. Will do it next week.

**Pull request merged**